apps/sms/test/marionette/composer_test.js

```diff
@@ -42,6 +42,57 @@ marionette('Messages Composer', function() {
     );
   });
 
+  suite('Preserve message input while navigating', function() {
+    var composer, threadList, thread;
+    var message = 'test message';
+
+    function waitForThreadList() {
+      client.helper.waitForElement(threadList.mmsThread);
+    }
+
+    function createMMSThread() {
+      threadList.navigateToComposer();
+      messagesApp.addRecipient('a@b.c');
+      messagesApp.addRecipient('s@p.c');
+      composer.messageInput.sendKeys('MMS thread.');
+      messagesApp.send();
+      messagesApp.performHeaderAction();
+    }
+
+    setup(function() {
+      thread = messagesApp.Thread;
+      composer = messagesApp.Composer;
+      threadList = messagesApp.ThreadList;
+
+      client.contentScript.inject(
```

Comment:
nit: please, move this to the upper-level setup so that all client script injections located in the same place.